src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -12,45 +12,86 @@
 using CilStrip.Mono.Cecil.Binary;
 using CilStrip.Mono.Cecil.Cil;
 using CilStrip.Mono.Cecil.Metadata;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using System.Buffers;
 
 public class ILStrip : Microsoft.Build.Utilities.Task
 {
     /// <summary>
     /// Assemblies to be stripped.
     /// The assemblies will be modified in place if OutputPath metadata is not set.
     /// </summary>
-    [Required]
     public ITaskItem[] Assemblies { get; set; } = Array.Empty<ITaskItem>();
 
     /// <summary>
     /// Disable parallel stripping
     /// </summary>
     public bool DisableParallelStripping { get; set; }
 
+    /// <summary>
+    /// Enable the feature of trimming indiviual methods
+    /// </summary>
+    public bool TrimIndividualMethods { get; set; }
+
+    /// <summary>
+    /// Methods to be trimmed, identified by method token
+    /// </summary>
+    public ITaskItem[] MethodTokenFiles { get; set; } = Array.Empty<ITaskItem>();
+
     public override bool Execute()
     {
-        if (Assemblies.Length == 0)
+        if (!TrimIndividualMethods)
         {
-            throw new ArgumentException($"'{nameof(Assemblies)}' is required.", nameof(Assemblies));
-        }
+            if (Assemblies.Length == 0)
```

Comment:
I think it would be better to make a separate function and keep the `Execute` method tidy:

```csharp
public override bool Execute()
{
  if (!TrimIndividualMethods)
  {
    return StripWholeAssemblies();
  }
  else
  {
    return ZeroOutIndividualMethods();
  }
}